### ISSUE TITLE
Wazuh API Connection Error after container restart

### ISSUE THREAD

**Issue by abasinger** [wazuh/wazuh-kibana-app#2439]
I apologize up front as I am new to docker and Wazuh and am learning fast. 

With a fresh docker pull of Wazuh with ELK on a fresh Linux install it works and seems to be good right out of the box. 

I can connect clients and get data, but if I make a change that requires a container restart I immediately get the API error from Wazuh. 

I am not finding any good data when it comes to troubleshooting the container versions as most of the data for troubleshooting this is not directed at them.

So things like running curl needs root permissions but I am unable to find the root user or password on the containers in any docks. Only password provided is of course foo bar which right now my lab is :)

**Comment by abasinger** [comment]
Additional information I did check the following "curl --basic -u foo:bar http://wazuh:55000?pretty

getting failed to connect to wazuh connection refused.

**Comment by abasinger** [comment]
Also checked The file wazuh.yml in the path: /usr/share/kibana/optimize/wazuh/config/wazuh.yml

what is interesting is the it is pointing to https://wazuh and the foo bar user and password are being used. 
So how is it resolving the name as I am unable to dig the name from inside the container? 
So why did it change from default install when restarted??? 

BTW I have installed the default 3.12. container installs and tested and it breaks every one when you restart your containers.

**Comment by jsanchez91** [comment]
Hi @abasinger 

To help you I would need to know which docker image you are using and describe the steps to reproduce the error. Tell me what changes you make and which containers.

**Comment by abasinger** [comment]
Hello @jsanchez91  

I installed 3.12.2_7.6.2 and to reproduce the issue in all 3 instances all I did was restart the containers with the docker restart (container) command. Then restarted the linux host. Debian 4.19.0-9-amd64.

The linux VMs running in Proxmox needed more memory which is why I shut the containers down then restarted linux and when it came back up wazuh was broken. 

I do see kibana is still working and data is still being fed into the ELK stack from the agents, but wazuh app is unable to talk to the API.

In looking at the API settings I see it (this is after reboot) it set to https://wazuh.

So basically I changed nothing after installation, then I deployed agents, and after getting 150 agents deployed needed memory added to the host, which needed a reboot, shut the containers down, rebooted and when it all came up wazuh did not work. I have reproduced this issue 3 times with 3 different fresh docker installations in VMs on Proxmox

**Comment by abasinger** [comment]
Sorry missed this Docker version 19.03.12

**Comment by jsanchez91** [comment]
Hello @abasinger,

Could you run this command on the wazuh manager's container?
```
curl -k -u foo:bar https://localhost:55000
```

On the other hand, could you share with me your Dockerfiles so that I try to reproduce your mistake in my environment?

**Comment by abasinger** [comment]
After running docker exc -it (wazuh manager container) /bin/bash

I ran curl -k -u foo:bar https://localhost:55000

Here is the output: Failed to connect localhost port 55000 connection refused

**Comment by abasinger** [comment]
If the connection is rejected, check the value of the host and port configured in the /var/ossec/api/configuration/api.yml

This file does not exist in the wazuh container in this location only config.js and preload_vars.conf and an auth folder and ssl folder

**Comment by abasinger** [comment]
sorry again am really new with Docker so tell me the best way to share the files and I will.

**Comment by jsanchez91** [comment]
Hi @abasinger,

I'm going to try to create an environment using Promox and version 3.12.2_7.6.2 to replicate your error. I'll give you an answer as soon as possible.

**Comment by jsanchez91** [comment]
I'm sorry, I couldn't reproduce your error.

I would need you to share with me the output of executing the following commands in a Wazuh manager:
1: Wazuh manager status
```
service wazuh-manager status
```

2: Wazuh manager log
```
tail -n 50 /var/ossec/logs/ossec.log
```

3: Wazuh API status
```
service wazuh-api status
```

4: Wazuh API log
```
tail -n 50 /var/ossec/logs/api.log
```

**Comment by abasinger** [comment]
Step 1.) output
root@wazuh-manager:/# service wazuh-manager status
wazuh-clusterd is running...
wazuh-modulesd is running...
ossec-monitord is running...
ossec-logcollector is running...
ossec-remoted is running...
ossec-syscheckd is running...
ossec-analysisd is running...
ossec-maild not running...
ossec-execd is running...
wazuh-db is running...
ossec-authd is running...
ossec-agentlessd not running...
ossec-integratord not running...
ossec-dbd not running...
ossec-csyslogd not running...
root@wazuh-manager:/# 


Step 2.) output

root@wazuh-manager:/# tail -n 50 /var/ossec/logs/ossec.log
2020/08/25 17:36:32 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '112' at 'any'.
2020/08/25 17:36:32 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '216.177.181.6' (name 'unknown').
2020/08/25 17:36:33 ossec-remoted: WARNING: (1408): Invalid ID 057 for the source ip: '74.195.123.142' (name 'unknown').
2020/08/25 17:36:33 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '118' at 'any'.
2020/08/25 17:36:33 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '009' at '192.168.111.200'.
2020/08/25 17:36:33 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '080' at 'any'.
2020/08/25 17:36:34 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '121' at 'any'.
2020/08/25 17:36:34 ossec-remoted: WARNING: (1408): Invalid ID 040 for the source ip: '72.47.35.121' (name 'unknown').
2020/08/25 17:36:34 ossec-remoted: WARNING: (1408): Invalid ID 105 for the source ip: '50.84.106.186' (name 'unknown').
2020/08/25 17:36:34 ossec-remoted: WARNING: (1408): Invalid ID 081 for the source ip: '24.27.6.16' (name 'unknown').
2020/08/25 17:36:34 ossec-remoted: WARNING: (1408): Invalid ID 194 for the source ip: '71.40.104.242' (name 'unknown').
2020/08/25 17:36:35 ossec-remoted: WARNING: (1408): Invalid ID 091 for the source ip: '71.40.215.62' (name 'unknown').
2020/08/25 17:36:35 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '010' at 'any'.
2020/08/25 17:36:35 ossec-remoted: WARNING: (1408): Invalid ID 025 for the source ip: '67.48.39.118' (name 'unknown').
2020/08/25 17:36:35 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '115' at 'any'.
2020/08/25 17:36:35 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '008' at 'any'.
2020/08/25 17:36:35 ossec-remoted: WARNING: (1408): Invalid ID 070 for the source ip: '67.78.72.174' (name 'unknown').
2020/08/25 17:36:35 ossec-remoted: WARNING: (1408): Invalid ID 061 for the source ip: '74.195.123.142' (name 'unknown').
2020/08/25 17:36:35 ossec-remoted: WARNING: (1408): Invalid ID 086 for the source ip: '71.40.215.62' (name 'unknown').
2020/08/25 17:36:36 ossec-remoted: WARNING: (1408): Invalid ID 073 for the source ip: '24.27.6.16' (name 'unknown').
2020/08/25 17:36:36 ossec-remoted: WARNING: (1408): Invalid ID 075 for the source ip: '24.27.6.16' (name 'unknown').
2020/08/25 17:36:36 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '117' at 'any'.
2020/08/25 17:36:36 ossec-remoted: WARNING: (1408): Invalid ID 004 for the source ip: '216.177.181.6' (name 'unknown').
2020/08/25 17:36:37 ossec-remoted: WARNING: (1408): Invalid ID 057 for the source ip: '74.195.123.142' (name 'unknown').
2020/08/25 17:36:37 ossec-remoted: WARNING: (1408): Invalid ID 006 for the source ip: '216.177.181.6' (name 'unknown').
2020/08/25 17:36:38 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '116' at 'any'.
2020/08/25 17:36:38 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '119' at 'any'.
2020/08/25 17:36:38 ossec-remoted: WARNING: (1408): Invalid ID 105 for the source ip: '50.84.106.186' (name 'unknown').
2020/08/25 17:36:38 ossec-remoted: WARNING: (1408): Invalid ID 081 for the source ip: '24.27.6.16' (name 'unknown').
2020/08/25 17:36:38 ossec-remoted: WARNING: (1408): Invalid ID 194 for the source ip: '71.40.104.242' (name 'unknown').
2020/08/25 17:36:38 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '216.177.181.6' (name 'unknown').
2020/08/25 17:36:40 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '080' at 'any'.
2020/08/25 17:36:40 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '121' at 'any'.
2020/08/25 17:36:40 ossec-remoted: WARNING: (1408): Invalid ID 040 for the source ip: '72.47.35.121' (name 'unknown').
2020/08/25 17:36:40 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '010' at 'any'.
2020/08/25 17:36:40 ossec-remoted: WARNING: (1408): Invalid ID 025 for the source ip: '67.48.39.118' (name 'unknown').
2020/08/25 17:36:41 ossec-remoted: WARNING: (1408): Invalid ID 091 for the source ip: '71.40.215.62' (name 'unknown').
2020/08/25 17:36:41 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '115' at 'any'.
2020/08/25 17:36:41 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '008' at 'any'.
2020/08/25 17:36:41 ossec-remoted: WARNING: (1408): Invalid ID 070 for the source ip: '67.78.72.174' (name 'unknown').
2020/08/25 17:36:42 ossec-remoted: WARNING: (1408): Invalid ID 086 for the source ip: '71.40.215.62' (name 'unknown').
2020/08/25 17:36:42 ossec-remoted: WARNING: (1408): Invalid ID 057 for the source ip: '74.195.123.142' (name 'unknown').
2020/08/25 17:36:42 ossec-remoted: WARNING: (1408): Invalid ID 073 for the source ip: '24.27.6.16' (name 'unknown').
2020/08/25 17:36:42 ossec-remoted: WARNING: (1408): Invalid ID 058 for the source ip: '74.195.123.142' (name 'unknown').
2020/08/25 17:36:42 ossec-remoted: WARNING: (1408): Invalid ID 075 for the source ip: '24.27.6.16' (name 'unknown').
2020/08/25 17:36:42 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '117' at 'any'.
2020/08/25 17:36:42 ossec-remoted: WARNING: (1408): Invalid ID 005 for the source ip: '216.177.181.6' (name 'unknown').
2020/08/25 17:36:42 ossec-remoted: WARNING: (1408): Invalid ID 004 for the source ip: '216.177.181.6' (name 'unknown').
2020/08/25 17:36:43 ossec-remoted: WARNING: (1408): Invalid ID 105 for the source ip: '50.84.106.186' (name 'unknown').
2020/08/25 17:36:43 ossec-remoted: WARNING: (1408): Invalid ID 081 for the source ip: '24.27.6.16' (name 'unknown').


Step 3.) output
root@wazuh-manager:/# service wazuh-api status
WAZUH-API is stopped.

Last 20 log entries:
WazuhAPI 2020-08-17 18:09:31 foo: [::ffff:172.21.0.4] GET /summary/agents? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:32 foo: [::ffff:172.21.0.4] GET /agents/summary? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:32 foo: [::ffff:172.21.0.4] GET /agents/000/config/auth/auth? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:02 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=1&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:04 foo: [::ffff:172.21.0.4] GET /cluster/status? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:06 foo: [::ffff:172.21.0.4] GET /cluster/node? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:09 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=500&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:40 foo: [::ffff:172.21.0.4] GET /manager/info? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:41 foo: [::ffff:172.21.0.4] GET /version? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:43 foo: [::ffff:172.21.0.4] GET /summary/agents? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:43 foo: [::ffff:172.21.0.4] GET /agents/summary? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:44 foo: [::ffff:172.21.0.4] GET /agents/000/config/auth/auth? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:02 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=1&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:04 foo: [::ffff:172.21.0.4] GET /cluster/status? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:06 foo: [::ffff:172.21.0.4] GET /cluster/node? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:09 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=500&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:12 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=1&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:14 foo: [::ffff:172.21.0.4] GET /cluster/status? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:16 foo: [::ffff:172.21.0.4] GET /cluster/node? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:19 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=500&q=id!%3D000 - 200 - error: '0'.

Step 4.) Output

root@wazuh-manager:/# tail -n 50 /var/ossec/logs/api.log
WazuhAPI 2020-08-17 18:07:43 foo: [::ffff:172.21.0.4] GET /agents/?limit=500&offset=0&select=id%2Cname - 200 - error: '0'.
WazuhAPI 2020-08-17 18:07:49 foo: [::ffff:172.21.0.4] GET /agents/?limit=500&offset=0&select=id%2Cname&search=DESKTOP-AM7ONE9 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:07:55 foo: [::ffff:172.21.0.4] POST /agents/group/Simmons - 200 - error: '0'.
WazuhAPI 2020-08-17 18:07:57 foo: [::ffff:172.21.0.4] GET /agents/groups/Simmons?limit=1 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:07:57 foo: [::ffff:172.21.0.4] GET /agents/groups/Simmons?limit=500&offset=0 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:00 foo: [::ffff:172.21.0.4] GET /agents/groups?search=Simmons - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:10 foo: [::ffff:172.21.0.4] GET /manager/info? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:13 foo: [::ffff:172.21.0.4] GET /summary/agents? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:13 foo: [::ffff:172.21.0.4] GET /version? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:15 foo: [::ffff:172.21.0.4] GET /agents/?limit=500&offset=0&sort=id - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:15 foo: [::ffff:172.21.0.4] GET /agents/000/config/auth/auth? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:20 foo: [::ffff:172.21.0.4] GET /agents/?limit=500&offset=0&sort=-status - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:25 foo: [::ffff:172.21.0.4] GET /agents/?limit=500&offset=0&sort=status - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:34 foo: [::ffff:172.21.0.4] GET /manager/info? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:37 foo: [::ffff:172.21.0.4] GET /agents/166? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:43 foo: [::ffff:172.21.0.4] GET /agents/groups? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:52 foo: [::ffff:172.21.0.4] GET /manager/info? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:56 foo: [::ffff:172.21.0.4] GET /cluster/nodes? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:08:56 foo: [::ffff:172.21.0.4] GET /agents/groups?limit=500&offset=0 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:03 foo: [::ffff:172.21.0.4] GET /agents/groups/Litton/files?limit=1 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:06 foo: [::ffff:172.21.0.4] GET /agents/groups/Litton?limit=500&offset=0 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:07 foo: [::ffff:172.21.0.4] GET /agents/groups/Litton?offset=0&select=id%2Cname - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:08 foo: [::ffff:172.21.0.4] GET /agents/?limit=500&offset=0&select=id%2Cname - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:08 foo: [::ffff:172.21.0.4] GET /agents/?limit=500&offset=0&select=id%2Cname&search=DESKTOP-AFDG0A4 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:22 foo: [::ffff:172.21.0.4] POST /agents/group/Litton - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:25 foo: [::ffff:172.21.0.4] GET /agents/groups/Litton?limit=1 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:25 foo: [::ffff:172.21.0.4] GET /agents/groups/Litton?limit=500&offset=0 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:26 foo: [::ffff:172.21.0.4] GET /agents/groups?search=Litton - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:29 foo: [::ffff:172.21.0.4] GET /manager/info? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:30 foo: [::ffff:172.21.0.4] GET /version? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:31 foo: [::ffff:172.21.0.4] GET /summary/agents? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:32 foo: [::ffff:172.21.0.4] GET /agents/summary? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:09:32 foo: [::ffff:172.21.0.4] GET /agents/000/config/auth/auth? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:02 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=1&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:04 foo: [::ffff:172.21.0.4] GET /cluster/status? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:06 foo: [::ffff:172.21.0.4] GET /cluster/node? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:15:09 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=500&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:40 foo: [::ffff:172.21.0.4] GET /manager/info? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:41 foo: [::ffff:172.21.0.4] GET /version? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:43 foo: [::ffff:172.21.0.4] GET /summary/agents? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:43 foo: [::ffff:172.21.0.4] GET /agents/summary? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:20:44 foo: [::ffff:172.21.0.4] GET /agents/000/config/auth/auth? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:02 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=1&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:04 foo: [::ffff:172.21.0.4] GET /cluster/status? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:06 foo: [::ffff:172.21.0.4] GET /cluster/node? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:30:09 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=500&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:12 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=1&q=id!%3D000 - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:14 foo: [::ffff:172.21.0.4] GET /cluster/status? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:16 foo: [::ffff:172.21.0.4] GET /cluster/node? - 200 - error: '0'.
WazuhAPI 2020-08-17 18:45:19 foo: [::ffff:172.21.0.4] GET /agents/?offset=0&limit=500&q=id!%3D000 - 200 - error: '0'.
root@wazuh-manager:/# 


Looks like the API is not even running

**Comment by abasinger** [comment]
ran the service wazuh-api start command and it started and is back up? What can I do to make sure the API starts on reset of the containers??

Thanks,

**Comment by jsanchez91** [comment]
In theory the Wazuh-API service should start with the container and the log does not show any error that prevented it from starting.

After starting the API manually, if you restart the container it will be stopped again?

**Comment by abasinger** [comment]
I will try that today and let you know.


I do have a question about the update services in wazuh to update definitions. Should I open a new thread?

I tried to run the command and it complained about 3.12 said I needed to be at 3.13? This should not be an expected behavior as most people will never be at the most recent version and forcing people to update will cause more issues.

Next I went over the update process and it is pretty straight forward except for the agent update. It does not tell you where your updating agents after updating the manager and API.
________________________________

**Comment by frankeros** [comment]
Due to inactivity, I'm closing this issue. Feel free to reopen it. If you have any questions, you can also use our Slack channel or subscribe to our mailing list.
